src/main/java/io/github/ibuildthecloud/gdapi/validation/ResourceManagerReferenceValidator.java

```diff
@@ -37,6 +37,24 @@ public Object getByField(String type, String fieldName, Object value) {
         return list.size() == 0 ? null : list.get(0);
     }
 
+    @Override
+    public Object getByFieldExcludeById(String type, String fieldName, Object value, String id) {
```

Comment:
Can you make this not largely copy and paste code.  So add on common method and if `id ==null` then do `getByField` behavior and if `id != null` do `getByFieldExcludedById` behavior